Libraries/plist/Tools/PlistBuddy.cpp

```diff
@@ -499,7 +499,14 @@ Merge(plist::Object *object, std::string const &mergeSource, std::queue<std::str
 
     if (auto targetDictionary = plist::CastTo<plist::Dictionary>(target)) {
         if (auto mergeObjectDictionary = plist::CastTo<plist::Dictionary>(mergeObject.get())) {
-            targetDictionary->merge(mergeObjectDictionary);
+            // Check to see if duplicate keys exists between target and merge dictionary,
+            // and print warning that those keys will be skipped.
+            for (auto const &mergeKey : *mergeObjectDictionary) {
+                if (targetDictionary->value(mergeKey)) {
+                    fprintf(stderr, "Skipping duplicate key: %s\n", mergeKey.c_str());
```

Comment:
You print a warning here, but you don't actually skip the key. Unless the second argument to `targetDictionary->merge()` indicates that?

Also, this seems inefficient. An easy thing to do is to `break;` the loop after printing the warning. Another thing is that you could return a boolean from the `merge()` method and only do additional checking (including the loop here) only if that method returns `false` for instance.